xdsl/printer.py

```diff
@@ -34,6 +34,9 @@ class Target(Enum):
 
     _indent: int = field(default=0, init=False)
     _ssa_values: Dict[SSAValue, str] = field(default_factory=dict, init=False)
+    """
+    maps SSA Values to their "allocated" names
```

Comment:
Is this docu for above or below? I think we usually document above fields, right?

Could you add docu to `_ssa_names` (number of occurences)? I think this is extremely purely named, I had to rethink how this works a couple of times until i got it again. It will certainly benefit from docu :sweat_smile:

Reply:
Docstrings always attach to the "above" thingy, which is quite confusing when commenting on attributes. I'll open a PR soon which adds a lot more docstrings here, just don't have the time for it right now :D